source/common/http/utility.cc

```diff
@@ -492,7 +492,7 @@ void Utility::transformUpgradeRequestFromH1toH2(HeaderMap& headers) {
 
 void Utility::transformUpgradeResponseFromH1toH2(HeaderMap& headers) {
   if (getResponseStatus(headers) == 101) {
-    headers.insertStatus().value().setInteger(200);
+    headers.setStatus(200);
```

Comment:
I think maybe Lizan suggested already -- we probably don't need integer setters for every builtin header. Maybe just status and contentLength, which we could just add in explicitly as helpers.


That might (or might not) help compile/link speed :)

Also for Status we could use the http code enum rather than an int, and for both status and content-length we could maybe sanity check the input with ASSERTs.

Let's just remember that for the next round.

Reply:
Yeah I suggested that but it's fine to leave it for next round.

Reply:
+1 to all those suggestions and also fine with me to defer to the next PR.